carbon-route.html

```diff
@@ -161,6 +161,23 @@
       this.linkPaths('tail.__queryParams', 'route.__queryParams');
     },
 
+    // IE Object.assign polyfill
+    __assign: function(target) {
+      target = Object(target);
+
+      for (var i = 1; i < arguments.length; i++) {
```

Comment:
`arguments` can be hard for some JS engines to do quickly, and we're only ever calling this with 2 args, so it's probably worth it to just make source an explicit parameter and drop the use of `arguments`